### ISSUE TITLE
Using UiKit with Next.js - ReferenceError: window is not defined

### ISSUE THREAD

**Issue by Brlaney** [uikit/uikit#4592]
### UIkit Version

In my *package.json* file:
```javascript
"uikit": "^3.13.1"
```

### Actual Behavior

The following error is displayed in my browser and in my terminal:

```bash
C:\Users\USER\Documents\Coding\portfolio\next-ts>yarn dev

yarn run v1.22.15
$ next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
event - compiled client and server successfully in 465 ms (405 modules)
wait  - compiling /_error (client and server)...
event - compiled client and server successfully in 132 ms (406 modules)

ReferenceError: window is not defined
    at C:\Users\USER\Documents\Coding\portfolio\next-ts\node_modules\uikit\dist\js\uikit.js:1481:31
    at C:\Users\USER\Documents\Coding\portfolio\next-ts\node_modules\uikit\dist\js\uikit.js:4:85
    at Object.<anonymous> (C:\Users\USER\Documents\Coding\portfolio\next-ts\node_modules\uikit\dist\js\uikit.js:7:3)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.uikit/dist/js/uikit.js (C:\Users\USER\Documents\Coding\portfolio\next-ts\.next\server\pages\_app.js:413:18)

wait  - compiling / (client and server)...
event - compiled client and server successfully in 1105 ms (412 modules)

ReferenceError: window is not defined
    at C:\Users\USER\Documents\Coding\portfolio\next-ts\node_modules\uikit\dist\js\uikit.js:1481:31
    at C:\Users\USER\Documents\Coding\portfolio\next-ts\node_modules\uikit\dist\js\uikit.js:4:85
    at Object.<anonymous> (C:\Users\USER\Documents\Coding\portfolio\next-ts\node_modules\uikit\dist\js\uikit.js:7:3)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.uikit/dist/js/uikit.js (C:\Users\USER\Documents\Coding\portfolio\next-ts\.next\server\pages\_app.js:413:18) {
  page: '/'
}
```

### Expected Behavior

I've been developing this project for a few weeks now and started getting this error today. The error started occurring after I removed 3 unused dependencies from my package.json file (using `yarn remove <package-name>`). 

Prior to today, the project would run just fine after running `yarn dev`.

It appears that the source of my error is from:
- node_modules\uikit\dist\js\uikit.js:1481:31
- node_modules\uikit\dist\js\uikit.js:4:85
- node_modules\uikit\dist\js\uikit.js:7:3

### Reproduction Link

I don't typically work with Codepen so hopefully, this will suffice: [Project Repo](https://github.com/Brlaney/next-ts.git)

### Steps to Reproduce the Problem

1. Run: `git clone https://github.com/Brlaney/next-ts.git` and `cd next-ts`
2. Run: `npm install` or (what I use) `yarn install`
3. Lastly, run `npm run dev` or `yarn dev`  and open **http://localhost:3000** in your browser. The error message I am receiving should be displayed in your browser and in your terminal.

**Comment by Brlaney** [comment]
I wanted to provide a brief update, I'm not currently using any of the UIkit javascript in my project so my current workaround is to just comment out my uikit.js import from my _app.tsx file (see the attached image).

I thought a solution would be to use Next.js's dynamic import feature, but that didn't seem to fix the issue. I believe this issue began yesterday because of the new release of UIkit, but I am still unsure as to what exactly is causing the issue. 

![update 3_19_22](https://user-images.githubusercontent.com/64326462/159133381-260ffb8e-ea6e-4894-9b19-2c8ac58ec635.png)

**Comment by shenst1** [comment]
Same issue. I pinned my UIkit version to an older version (3.6.17) until this issue is resolved. There are probably much newer working versions but that one worked fine.

**Comment by Luukuton** [comment]
Same issue here. [Upgraded UIKit](https://github.com/Luukuton/shingetsu-blog/commit/a41ee88d0903e2d359bf919a3be617fac06846c6) from 3.11.1 to 3.13.1 which caused the error.

Loading UIKit's JS from the CDN in **_document.tsx** works as a workaround.

```tsx
<Head>
  {/* ... */}
  <script defer src="https://cdn.jsdelivr.net/npm/uikit@3.13.1/dist/js/uikit.min.js" />
</Head>
```

**Comment by BossBele** [comment]
Workaround for UIkit 3.16.6 with Next JS:

In _app.js:

```
import 'uikit/dist/css/uikit.min.css';

import { useCallback, useEffect } from 'react';
import Icons from 'uikit/dist/js/uikit-icons';

export default function App({ Component, pageProps }) {
  const importUIkit = useCallback(async() => {
    const UIkit = (await import('uikit')).default;
    UIkit.use(Icons);
  },[]);

  useEffect(() => {
    importUIkit();
  },[importUIkit]);

  return (
      <Component {...pageProps} />
  );
}
```

**Comment by nitras** [comment]
> 

I am unsure what I do wrong, but this does not work for me.

`{
	"name": "nextjs-uikit-starter",
	"version": "0.1.0",
	"private": true,
	"scripts": {
		"dev": "next dev",
		"build": "next build",
		"start": "next start",
		"lint": "next lint"
	},
	"dependencies": {
		"next": "13.2.4",
		"react": "18.2.0",
		"react-dom": "18.2.0",
		"uikit": "^3.16.6"
	}
}
`

**Comment by BossBele** [comment]
@nitras please share what your pages/_app.js looks like

**Comment by nitras** [comment]
> @nitras please share what your pages/_app.js looks like


The same as what you posted.

`import '@/styles/globals.css';

import { useCallback, useEffect } from 'react';
import Icons from 'uikit/dist/js/uikit-icons';

export default function App({ Component, pageProps }) {
	const importUIkit = useCallback(async () => {
		const UIkit = (await import('uikit')).default;
		UIkit.use(Icons);
	}, []);

	useEffect(() => {
		importUIkit();
	}, [importUIkit]);

	return <Component {...pageProps} />;
}
`

**Comment by BossBele** [comment]
@nitras Here's a working sandbox to help you see how it works:
https://codesandbox.io/p/sandbox/strange-cannon-7d7h1e?file=%2Fpages%2Findex.tsx&selection=%5B%7B%22endColumn%22%3A34%2C%22endLineNumber%22%3A32%2C%22startColumn%22%3A34%2C%22startLineNumber%22%3A32%7D%5D

Here's the generated page/output: (You'll notice two check marks/icons at the top which are UIKit Icons)

https://7d7h1e-3000.csb.app/

**Comment by janschoenherr** [comment]
This should be fixed with 3.16.7 again. Thanks.
